lib/utils/batcher/state.go

```diff
@@ -94,6 +114,14 @@ func (sm *StateMonitor) UpdateState(batchSize int) State {
 	newState := sm.evaluateState(batchSize)
 	sm.state = newState
 	sm.mu.Unlock()
+	switch newState {
+	case StateOverloaded:
+		if sm.OverloadedIdle > 0 {
+			sm.initIdleTimeoutOnOverloadState()
+		}
```

Comment:
I don't have an issue with the approach of resetting the state to normal asynchronously but I do wonder if there's a simple way to implement this. The current logic tries to schedule reset after we've moved to the overloaded state which was a bit hard to follow for me.

Would it be simpler if this StateMonitor just always ran a background goroutine that does this:

- Wakes up each "OverloadIdleTimeout" and checks when was the last state change.
- If it was longer ago than "OverloadIdleTimeout", change it back to normal.

We'd need to keep track of the state change time but that's easy (can just update it above in this function), but then the "reset" logic would be all inside a routine so it would be easy to follow.

WDYT?